iml-manager-cli/src/api.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlManagerCliError;
+use console::Term;
+use structopt::{clap::arg_enum, StructOpt};
+
+arg_enum! {
+    #[derive(PartialEq, Debug)]
+    pub enum ApiMethod {
+        Delete,
+        Get,
+        Head,
+        Patch,
+        Post,
+        Put,
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct ApiCommand {
+    #[structopt(possible_values = &ApiMethod::variants(), case_insensitive = true)]
+    method: ApiMethod,
+
+    path: String,
+
+    /// JSON formatted body to send
+    #[structopt(required_ifs(&[("method", "patch"),("method", "post"),("method", "put")]))]
+    body: Option<String>,
+}
+
+pub async fn api_cli(command: ApiCommand) -> Result<(), ImlManagerCliError> {
+    let term = Term::stdout();
+    let client = iml_manager_client::get_client()?;
+    let uri = iml_manager_client::create_api_url(command.path)?;
+
+    let req = match command.method {
+        ApiMethod::Delete => client.delete(uri),
+        ApiMethod::Get => client.get(uri),
+        ApiMethod::Head => client.head(uri),
+        ApiMethod::Patch => client.patch(uri),
+        ApiMethod::Post => client.post(uri),
+        ApiMethod::Put => client.put(uri),
```

Comment:
Headers and bodies are not supported, is it intentional?

Reply:
Bodies are supported here: https://github.com/whamcloud/integrated-manager-for-lustre/pull/2132/files#diff-f33d40dcab0b4085abfc004fe3408e29R50-R54

Headers generally aren't needed (currently) when using the ```iml_manager_client``